pkg/reconcile/pipeline/api.go

```diff
@@ -22,6 +24,21 @@ type Pipeline interface {
 	Process(binding interface{}) (bool, error)
 }
 
+// A container-like object.  EnvFrom is optional; all other fields are required.
+type MetaContainer struct {
+	Name        string
+	Env         []string
+	EnvFrom     []string
+	VolumeMount []string
+	Data        map[string]interface{}
+}
+
+type MetaContainerTemplate struct {
+	Containers []MetaContainer
+	Volume     []string
+	Data       map[string]interface{}
```

Comment:
I use the `Data` field (and the other `Data` field in `MetaContainer`) in two places:
1. Unit tests.
2. `mapping.go`, where I need to get the unstructured form of the resource to write into it.

While I could probably get away with a getter method in `mapping.go`, it's convenient for the tests.  Is there a way of doing this without resorting to public use?  I'd like to avoid it if possible.